gax-java/gax/src/main/java/com/google/api/gax/paging/Page.java

```diff
@@ -52,12 +55,26 @@
   Page<ResourceT> getNextPage();
 
   /**
-   * Returns an iterable that traverses all of the elements of the underlying data source. The data
-   * is fetched lazily page by page, where each page may contain multiple elements. A new page is
+   * Returns an iterable that traverses all the elements of the underlying data source. The data is
+   * fetched lazily page by page, where each page may contain multiple elements. A new page is
    * fetched whenever the elements of any particular page are exhausted.
    */
   Iterable<ResourceT> iterateAll();
 
   /** Returns an iterable over the elements in this page. */
   Iterable<ResourceT> getValues();
+
+  /**
+   * Returns a stream that traverses all the elements of the underlying data source. The data is
+   * fetched lazily page by page, where each page may contain multiple elements. A new page is
+   * fetched whenever the elements of any particular page are exhausted.
+   */
+  default Stream<ResourceT> streamAll() {
+    return StreamSupport.stream(iterateAll().spliterator(), false);
```

Comment:
@BenWhitehead Would you elaborate what's not ideal? Parallel Stream.

Reply:
I think this is great for a first step of getting it on the api and saving our customers from having to do the same hoop jumping.

In the case of future improvements, right now this approach will not allow of parallel processing the same way a natively parallel friendly stream would. In all actuality, Page isn't fully parallelizable due to the fact that each page contains the `nextPageToken` necessary in order to fetch the next page. But, Page could be more parallel friendly where processing of the actual resources can happen on a thread separate from the thread performing the actual get.

Reply:
@JoeWang1127 Do you have question or comment about parallelism? (I don't think of an implementation as of now)

Reply:
I agree, parallelism does not need to block this PR and can be an additional improvement in the future.

Reply:
@suztomo I'm not understand about the separate thread in Ben's comment:
```
Page could be more parallel friendly where processing of the actual resources can happen on a thread separate from the thread performing the actual get.
```

Reply:
This https://docs.oracle.com/javase/tutorial/collections/streams/parallelism.html#executing_streams_in_parallel

Reply:
Thanks for the reference.

I don't have a question about it.